torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1592,17 +1592,15 @@ int listAdd(Stack& stack) {
   TList b;
   pop(stack, a, b);
 
-  std::vector<TElement> ret;
-  const auto total_size = a->elements().size() + b->elements().size();
+  auto& a_elements = a->elements();
+  auto& b_elements = b->elements();
+  const auto total_size = a_elements.size() + b_elements.size();
+
+  std::vector<TElement> ret = std::move(a->elements());
   ret.reserve(total_size);
-  for (const auto& a_element : a->elements()) {
-    ret.push_back(a_element);
-  }
-  for (const auto& b_element : b->elements()) {
-    ret.push_back(b_element);
-  }
+  std::move(b_elements.begin(), b_elements.end(), std::back_inserter(ret));
```

Comment:
Does moving all the elements of `a` and `b` create a use-after-move issue if `a` or `b` is accessed later?

Reply:
Is it possible for someone to have another reference to them?

Reply:
oh, yes it is. Add is a functional operation.